### ISSUE TITLE
withTamagui crashes next app in nx repo (worked in old version)

### ISSUE THREAD

**Issue by Mimameid** [tamagui/tamagui#1344]
### Current Behavior

With the following versions `nx@16.4.0`, `next@13.4.2` and `tamagui@1.36.3` the `next` app crashes on startup when serving in 'dev' mode. Works with older version of said dependencies `nx@16.1.3`, `next@13.3.0` and `tamagui@1.29.1`

I suspect `withTamagui` plugin from `@tamagui/next-plugin` is the culprit here.

### Expected Behavior

App shouldn't crash.

### Tamagui Version

```markdown
1.36.3
```

### Reproduction

```markdown
I added a reproduction repo: https://github.com/mimameid/nx-next-tamagui/

You can find the working version (with `nx@16.1.3`, `next@13.3.0` and `tamagui@1.29.1`) on the branch `old-versions`
```

**Comment by natew** [comment]
What is the error on crash? No error it just exits?

If you could try adding some logs in various areas and tracking down where it actually crashes that would be helpful. You can also try adding `DEBUG=*` or `DEBUG=tamagui` to get a better hint of where.

**Comment by Mimameid** [comment]
Hey Nate, we are kinda lost tbh. We spent already several hours on this and can't find a starting point. The app crashes pretty much right away. It seems to be an issue with the `config` or its resolution in the `next.config.j`. We tried to get a more specific error manipulating `server.impl.js`. We also tried to play around with some `nx` configs, but to no avail. If you have any idea, where we can get to the bottom of this, we would gladly help.

This is my console output for the `DEBUG=* npx nx run next-app:serve --verbose ` command:

```
> nx run next-app:serve:development


 >  NX   RunStart

   
{
  "meta": {
    "nxCloudVersion": "16.0.5"
  },
  "branch": null,
  "runGroup": "cbac6249e627a54b1913088fd24824bf167dd956",
  "ciExecutionId": null,
  "ciExecutionEnv": "",
  "hashes": [
    "3439918690030861540"
  ]
}

  follow-redirects options {
  maxRedirects: 21,
  maxBodyLength: Infinity,
  protocol: 'https:',
  path: '/nx-cloud/runs/start',
  method: 'POST',
  headers: [Object: null prototype] {
    Accept: 'application/json, text/plain, */*',
    'Content-Type': 'application/json',
    authorization: 'YTJiYzMzNWMtN2ZhNy00ZWE1LWE5YzUtYWZmYmI3OTJjZGY2fHJlYWQtd3JpdGU=',
    'User-Agent': 'axios/1.1.3',
    'Content-Length': '178',
    'Accept-Encoding': 'gzip, deflate, br'
  },
  agents: { http: undefined, https: undefined },
  auth: undefined,
  beforeRedirect: [Function: dispatchBeforeRedirect],
  beforeRedirects: { proxy: [Function: beforeRedirect] },
  hostname: 'cloud.nx.app',
  port: '',
  agent: undefined,
  nativeProtocols: {
    'http:': {
      _connectionListener: [Function: connectionListener],
      METHODS: [Array],
      STATUS_CODES: [Object],
      Agent: [Function],
      ClientRequest: [Function: ClientRequest],
      IncomingMessage: [Function: IncomingMessage],
      OutgoingMessage: [Function: OutgoingMessage],
      Server: [Function: Server],
      ServerResponse: [Function: ServerResponse],
      createServer: [Function: createServer],
      validateHeaderName: [Function: __node_internal_],
      validateHeaderValue: [Function: __node_internal_],
      get: [Function: get],
      request: [Function: request],
      setMaxIdleHTTPParsers: [Function: setMaxIdleHTTPParsers],
      maxHeaderSize: [Getter],
      globalAgent: [Getter/Setter]
    },
    'https:': {
      Agent: [Function: Agent],
      globalAgent: [Agent],
      Server: [Function: Server],
      createServer: [Function: createServer],
      get: [Function: get],
      request: [Function: request]
    }
  }
} +0ms
RunStart duration: 179
- ready started server on 0.0.0.0:4200, url: http://localhost:4200
- info Loaded env from /Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app/.env.local
Disabling static extraction in development mode for better HMR
2023-06-27T14:29:28.198Z next:jsconfig-paths-plugin tsconfig.json or jsconfig.json paths: {}
2023-06-27T14:29:28.198Z next:jsconfig-paths-plugin resolved baseUrl: /Users/ibrahimkeser/projects/nx-next-tamagui
Tamagui alias: {
  '@opentelemetry/api': 'next/dist/compiled/@opentelemetry/api',
  next: '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next',
  'styled-jsx/style$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/styled-jsx/style.js',
  'styled-jsx$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/styled-jsx/index.js',
  'private-next-pages/_app': [ 'next/dist/pages/_app.js' ],
  'private-next-pages/_error': [ 'next/dist/pages/_error.js' ],
  'private-next-pages/_document': [ 'next/dist/pages/_document.js' ],
  'private-next-app-dir': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app/app',
  'private-next-root-dir': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app',
  'private-dot-next': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app/.next',
  'unfetch$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/fetch/index.js',
  'isomorphic-unfetch$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/fetch/index.js',
  'whatwg-fetch$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/fetch/whatwg-fetch.js',
  'object-assign$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object-assign.js',
  'object.assign/auto': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/auto.js',
  'object.assign/implementation': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/implementation.js',
  'object.assign$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/index.js',
  'object.assign/polyfill': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/polyfill.js',
  'object.assign/shim': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/shim.js',
  url: '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/compiled/native-url/index.js',
  'private-next-rsc-action-validate': 'next/dist/build/webpack/loaders/next-flight-loader/action-validate',
  'private-next-rsc-action-client-wrapper': 'next/dist/build/webpack/loaders/next-flight-loader/action-client-wrapper',
  'private-next-rsc-action-proxy': 'next/dist/build/webpack/loaders/next-flight-loader/action-proxy',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/shared/lib/router/utils/resolve-rewrites.js': false,
  '@swc/helpers/_': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@swc/helpers/_',
  setimmediate: 'next/dist/compiled/setimmediate',
  '@tamagui/core/reset.css': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/core/reset.css',
  '@tamagui/core': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/core/dist/esm/index.js',
  '@tamagui/web': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/web/dist/esm/index.js',
  'react-native-svg': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/react-native-svg/dist/esm/index.js',
  'react/jsx-runtime.js': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-runtime.js',
  'react/jsx-runtime': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-runtime.js',
  'react/jsx-dev-runtime.js': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-dev-runtime.js',
  'react/jsx-dev-runtime': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-dev-runtime.js',
  'react-native$': 'react-native-web-lite',
  'react-native-web$': 'react-native-web-lite',
  '@testing-library/react-native': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/proxy-worm/index.js',
  '@gorhom/bottom-sheet$': '@gorhom/bottom-sheet',
  'react-native/Libraries/Renderer/shims/ReactFabric': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/proxy-worm/index.js'
}
2023-06-27T14:29:28.241Z next:jsconfig-paths-plugin tsconfig.json or jsconfig.json paths: {}
2023-06-27T14:29:28.241Z next:jsconfig-paths-plugin resolved baseUrl: /Users/ibrahimkeser/projects/nx-next-tamagui
Tamagui alias: {
  '@opentelemetry/api': 'next/dist/compiled/@opentelemetry/api',
  next: '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next',
  'styled-jsx/style$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/styled-jsx/style.js',
  'styled-jsx$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/styled-jsx/index.js',
  'private-next-pages/_app': [ 'next/dist/pages/_app.js' ],
  'private-next-pages/_error': [ 'next/dist/pages/_error.js' ],
  'private-next-pages/_document': [ 'next/dist/pages/_document.js' ],
  'private-next-app-dir': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app/app',
  'private-next-root-dir': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app',
  'private-dot-next': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app/.next',
  'private-next-rsc-action-validate': 'next/dist/build/webpack/loaders/next-flight-loader/action-validate',
  'private-next-rsc-action-client-wrapper': 'next/dist/build/webpack/loaders/next-flight-loader/action-client-wrapper',
  'private-next-rsc-action-proxy': 'next/dist/build/webpack/loaders/next-flight-loader/action-proxy',
  '@swc/helpers/_': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@swc/helpers/_',
  setimmediate: 'next/dist/compiled/setimmediate',
  '@tamagui/core/reset.css': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/core/reset.css',
  '@tamagui/core': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/core/dist/cjs/index.js',
  '@tamagui/web': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/web/dist/cjs/index.js',
  'react-native-svg': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/react-native-svg/dist/cjs/index.js',
  'react/jsx-runtime.js': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-runtime.js',
  'react/jsx-runtime': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-runtime.js',
  'react/jsx-dev-runtime.js': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-dev-runtime.js',
  'react/jsx-dev-runtime': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-dev-runtime.js',
  'react-native$': 'react-native-web-lite',
  'react-native-web$': 'react-native-web-lite',
  '@testing-library/react-native': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/proxy-worm/index.js',
  '@gorhom/bottom-sheet$': '@gorhom/bottom-sheet',
  'react-native/Libraries/Renderer/shims/ReactFabric': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/proxy-worm/index.js'
}
2023-06-27T14:29:28.245Z next:jsconfig-paths-plugin tsconfig.json or jsconfig.json paths: {}
2023-06-27T14:29:28.245Z next:jsconfig-paths-plugin resolved baseUrl: /Users/ibrahimkeser/projects/nx-next-tamagui
Tamagui alias: {
  'next/dist/client': 'next/dist/esm/client',
  'next/dist/shared': 'next/dist/esm/shared',
  'next/dist/pages': 'next/dist/esm/pages',
  'next/dist/lib': 'next/dist/esm/lib',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/client/link.js': 'next/dist/esm/client/link',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/client/image.js': 'next/dist/esm/client/image',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/client/script.js': 'next/dist/esm/client/script',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/client/router.js': 'next/dist/esm/client/router',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/shared/lib/head.js': 'next/dist/esm/shared/lib/head',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/shared/lib/dynamic.js': 'next/dist/esm/shared/lib/dynamic',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/pages/_document.js': 'next/dist/esm/pages/_document',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/pages/_app.js': 'next/dist/esm/pages/_app',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/client/components/navigation.js': 'next/dist/client/components/navigation',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/client/components/headers.js': 'next/dist/client/components/headers',
  '@opentelemetry/api': 'next/dist/compiled/@opentelemetry/api',
  next: '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next',
  'styled-jsx/style$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/styled-jsx/style.js',
  'styled-jsx$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/styled-jsx/index.js',
  'private-next-pages/_app': [ 'next/dist/esm/pages/_app.js' ],
  'private-next-pages/_error': [ 'next/dist/esm/pages/_error.js' ],
  'private-next-pages/_document': [ 'next/dist/esm/pages/_document.js' ],
  'private-next-app-dir': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app/app',
  'private-next-root-dir': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app',
  'private-dot-next': '/Users/ibrahimkeser/projects/nx-next-tamagui/packages/next-app/.next',
  'unfetch$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/fetch/index.js',
  'isomorphic-unfetch$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/fetch/index.js',
  'whatwg-fetch$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/fetch/whatwg-fetch.js',
  'object-assign$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object-assign.js',
  'object.assign/auto': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/auto.js',
  'object.assign/implementation': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/implementation.js',
  'object.assign$': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/index.js',
  'object.assign/polyfill': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/polyfill.js',
  'object.assign/shim': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/build/polyfills/object.assign/shim.js',
  url: '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/compiled/native-url/index.js',
  'private-next-rsc-action-validate': 'next/dist/build/webpack/loaders/next-flight-loader/action-validate',
  'private-next-rsc-action-client-wrapper': 'next/dist/build/webpack/loaders/next-flight-loader/action-client-wrapper',
  'private-next-rsc-action-proxy': 'next/dist/build/webpack/loaders/next-flight-loader/action-proxy',
  '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/next/dist/shared/lib/router/utils/resolve-rewrites.js': false,
  '@swc/helpers/_': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@swc/helpers/_',
  setimmediate: 'next/dist/compiled/setimmediate',
  '@tamagui/core/reset.css': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/core/reset.css',
  '@tamagui/core': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/core/dist/cjs/index.js',
  '@tamagui/web': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/web/dist/cjs/index.js',
  'react-native-svg': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/react-native-svg/dist/cjs/index.js',
  'react/jsx-runtime.js': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-runtime.js',
  'react/jsx-runtime': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-runtime.js',
  'react/jsx-dev-runtime.js': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-dev-runtime.js',
  'react/jsx-dev-runtime': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/react/jsx-dev-runtime.js',
  'react-native$': 'react-native-web-lite',
  'react-native-web$': 'react-native-web-lite',
  '@testing-library/react-native': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/proxy-worm/index.js',
  '@gorhom/bottom-sheet$': '@gorhom/bottom-sheet',
  'react-native/Libraries/Renderer/shims/ReactFabric': '/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@tamagui/proxy-worm/index.js'
}
 >  NX   Next.js app exited with code 1
Error: Next.js app exited with code 1
    at ChildProcess.<anonymous> (/Users/ibrahimkeser/projects/nx-next-tamagui/node_modules/@nx/next/src/executors/server/server.impl.js:46:27)
    at Object.onceWrapper (node:events:628:26)
    at ChildProcess.emit (node:events:513:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)

 ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target serve for project next-app failed

   Failed tasks:
   
   - next-app:serve:development
   
   Hint: run the command with --verbose for more details.

   View structured, searchable error logs at https://nx.app/runs/eB1cyK62Xf
```

**Comment by moatorres** [comment]
I'm having the exact same issue. It's really hard to debug indeed. My hint is also how the "withTamagui" plugins interacts with "@nx/next" and/or the "composePlugins" helper. I've spent about 5 hours trying to figure out what's going on to no avail. I'll add more here if/when I manage to make it work.

@Mimameid have you managed to pinpoint which version introduced the bug? I'm trying to narrow it down the last working combination of libraries since the three had recent updates so I can analyse relevant changes and possible culprits.

**Comment by moatorres** [comment]
I'm having the exact same issue. It's really hard to debug indeed. My hint is also how the "withTamagui" interacts with "@nx/next" and/or the "composePlugins" helper. I've spent about 5 hours trying to figure out what's going on to no avail. I'll add more here if/when I manage to make it work. It could also be related to nx's default compiler update (swc) from your error logs. I don't think that's related to the underlying react-native deps though.

@Mimameid have you managed to pinpoint which version introduced the bug? I'm trying to narrow it down to the last working combination of libraries since the three had recent updates so I can analyse relevant changes and possible culprits.

**Comment by Mimameid** [comment]
@moatorres If I remember correctly the error occurs with the version `16.3.0` of `nx`. `nx` works with `next` together. But the moment one adds `withTamagui` to the mix, things break.

**Comment by kaceycleveland** [comment]
I was running into this while trying to create a generator for NX for NextJS to add tamagui automatically. I found I needed a `package.json` in the directory of the project which was interesting. Is there something in tamagui that looks for it?

**Comment by dfr-exnaton** [comment]
I can confirm that this resolves the issue!

**Comment by natew** [comment]
fix going out in 1.59.1
